src/org/opendatakit/briefcase/ui/reused/transfer/sourcetarget/CentralServerDialogForm.java

```diff
@@ -77,15 +93,32 @@
 
     connectButton.addActionListener(__ -> triggerConnect());
 
+    urlField.addFocusListener(new FocusAdapterBuilder()
+        .onFocusLost(e -> urlField.setText(CentralServer.cleanUrl(urlField.getText())))
+        .build());
+
+    urlValidator = UiFieldValidator.of(urlField, urlFieldLabel, REQUIRED, URI).onChange(this::updateConnectButton);
+    projectIdValidator = UiFieldValidator.of(projectIdField, projectIdFieldLabel, REQUIRED, NUMBER).onChange(this::updateConnectButton);
+    emailValidator = UiFieldValidator.of(emailField, emailFieldLabel, REQUIRED, EMAIL).onChange(this::updateConnectButton);
+    passwordValidator = UiFieldValidator.of(passwordField, passwordFieldLabel, REQUIRED).onChange(this::updateConnectButton);
+
     getRootPane().setDefaultButton(connectButton);
+  }
 
+  private void updateConnectButton() {
+    connectButton.setEnabled(Stream.of(
+        urlValidator.isValid(),
+        projectIdValidator.isValid(),
+        emailValidator.isValid(),
+        passwordValidator.isValid()
+    ).reduce(true, Boolean::logicalAnd));
   }
 
   private void triggerConnect() {
     setCursor(Cursor.getPredefinedCursor(Cursor.WAIT_CURSOR));
     try {
       Optional<Credentials> credentials = OptionalProduct.all(
-          Optional.ofNullable(usernameField.getText()).map(String::trim).filter(s -> !s.isEmpty()),
+          Optional.ofNullable(emailField.getText()).map(String::trim).filter(s -> !s.isEmpty()),
```

Comment:
At some point, shall we remove this duplication?

<img width="1537" alt="Screen Shot 2019-07-28 at 1 03 46 PM" src="https://user-images.githubusercontent.com/55099/62012236-91911480-b138-11e9-8f55-eaef65a9d933.png">